src/containers/register.js

```diff
@@ -1,15 +1,151 @@
 import React from "react";
 
-class RegisterContainer extends React.Component {
+import Button from "../components/Button";
+import Login from "../components/Login";
+import { Input, Row, Col } from "react-materialize";
+
+import { Link } from "react-router-dom";
+import { connect } from "react-redux";
+import linkState from "linkstate";
+
+import { LoginUser, RegisterCar } from "../actions/register";
+
+class Register extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      login: false,
+      register_car: true,
+      car: {
+        make: "",
+        model: "",
+        year: "",
+        color: ""
+      },
+      err: false
+    };
+
+    this.login = this.login.bind(this);
+    this.registerCar = this.registerCar.bind(this);
+  }
+
+  async login(token) {
+    const { err, data } = await this.props.login(token);
+    if (err) {
+      this.setState({ err });
+      return;
+    }
+    this.setState({
+      err: false,
+      login: false,
+      register_car: true
+    });
   }
+
+  async registerCar() {
+    const { err, car } = await this.props.registerCar(this.state.car);
+    if (err) {
+      this.setState({ err });
+      return;
+    }
+    await this.setState({
+      err: false,
+      login: false,
+      register_car: false
+    });
+    this.props.changePage("/dashboard");
+  }
+
   render() {
     const {} = this.props;
```

Comment:
does this destructuring work in regular react?

Reply:
yes